src/components/SearchView.tsx

```diff
@@ -19,30 +21,50 @@ interface SearchViewProps {
 interface SearchViewStates {
     searchText: string;
     displayMode: displayMode;
+    selectedCategories: string[];
+    structured: boolean;
+    detailed: boolean;
 }
 
 export class SearchView extends React.Component<SearchViewProps, SearchViewStates> {
     timeout: number;
 
     searchResultListItems: any;
+    categories: string[] = [];
 
     constructor(props: SearchViewProps) {
         super(props);
 
-        // set default state
-        this.state = ({
-            searchText: "",
-            displayMode: "list"
-        })
+        this.categories = ["List", "Input", "Math", "Script", "Display", "ImportExport", "String", "Geometry"];
```

Comment:
A `SearchView` component does not even care about these list, it's the caller's job.